analyzer/config/docker/docker.go

```diff
@@ -0,0 +1,65 @@
+package docker
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/open-policy-agent/conftest/parser/docker"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/fanal/analyzer"
+	"github.com/aquasecurity/fanal/analyzer/config"
+	"github.com/aquasecurity/fanal/types"
+)
+
+func init() {
+	analyzer.RegisterAnalyzer(&dockerConfigAnalyzer{
+		parser: &docker.Parser{},
+	})
+}
+
+const version = 1
+
+var requiredFile = "Dockerfile"
+
+type dockerConfigAnalyzer struct {
+	parser *docker.Parser
+}
+
+func (a dockerConfigAnalyzer) Analyze(target analyzer.AnalysisTarget) (*analyzer.AnalysisResult, error) {
+	var parsed interface{}
+	if err := a.parser.Unmarshal(target.Content, &parsed); err != nil {
+		return nil, xerrors.Errorf("unable to parse Dockerfile (%s): %w", target.FilePath, err)
+	}
+	return &analyzer.AnalysisResult{
+		Configs: []types.Config{{
+			Type:     config.Dockerfile,
+			FilePath: target.FilePath,
+			Content:  parsed,
+		}},
+	}, nil
+}
+
+// Required does a case-insensitive check for filePath and returns true if
+// filePath equals/startsWith/hasExtension requiredFile
+func (a dockerConfigAnalyzer) Required(filePath string, _ os.FileInfo) bool {
```

Comment:
I changed the entire checks to accept only `dockerfile.*` or `*.dockerfile`, with added test cases